File: backend/src/main/java/com/ndoruhirwe/smartlogistics/dto/response/PurchaseOrderResponse.java
package com.ndoruhirwe.smartlogistics.dto.response;

import com.ndoruhirwe.smartlogistics.entity.enums.PurchaseOrderStatus;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record PurchaseOrderResponse(

        UUID id,

        String orderNumber,

        UUID supplierId,

        String supplierCode,

        String supplierName,

        UUID warehouseId,

        String warehouseCode,

        String warehouseName,

        LocalDate orderDate,

        LocalDate expectedDeliveryDate,

        PurchaseOrderStatus status,

        String notes,

        List<PurchaseOrderItemResponse> items,

        BigDecimal totalAmount,

        LocalDateTime createdAt,

        LocalDateTime updatedAt
) {
}

File: frontend/src/app/features/procurement/models/purchase-order-response.ts
import { PurchaseOrderStatus } from './purchase-order-status';
import { PurchaseOrderItemResponse } from './purchase-order-item-response';

export interface PurchaseOrderResponse {
  id: string;

  orderNumber: string;

  supplierId: string;
  supplierCode: string;
  supplierName: string;

  warehouseId: string;
  warehouseCode: string;
  warehouseName: string;

  orderDate: string;
  expectedDeliveryDate: string | null;

  status: PurchaseOrderStatus;

  notes: string | null;

  items: PurchaseOrderItemResponse[];

  totalAmount: number;

  createdAt: string;
  updatedAt: string | null;
}
